cli/pkg/kctrl/cmd/app/delete.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	"github.com/spf13/cobra"
+	cmdcore "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/core"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/logger"
+	kcv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	kcclient "github.com/vmware-tanzu/carvel-kapp-controller/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type DeleteOptions struct {
+	ui          ui.UI
+	depsFactory cmdcore.DepsFactory
+	logger      logger.Logger
+
+	WaitFlags cmdcore.WaitFlags
+
+	NamespaceFlags cmdcore.NamespaceFlags
+	Name           string
+
+	IgnoreAssociatedResources bool
+}
+
+func NewDeleteOptions(ui ui.UI, depsFactory cmdcore.DepsFactory, logger logger.Logger) *DeleteOptions {
+	return &DeleteOptions{ui: ui, depsFactory: depsFactory, logger: logger}
+}
+
+func NewDeleteCmd(o *DeleteOptions, flagsFactory cmdcore.FlagsFactory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete App CR",
+		RunE:  func(_ *cobra.Command, _ []string) error { return o.Run() },
+	}
+
+	o.NamespaceFlags.Set(cmd, flagsFactory)
+	cmd.Flags().StringVarP(&o.Name, "app", "a", "", "Set App CR name (required)")
+	cmd.Flags().BoolVar(&o.IgnoreAssociatedResources, "ignore-associated-resources", false, "Ignore resources created by the AppCR and delete the custom resource itself")
+	o.WaitFlags.Set(cmd, flagsFactory, &cmdcore.WaitFlagsOpts{
+		AllowDisableWait: true,
+		DefaultInterval:  1 * time.Second,
+		DefaultTimeout:   5 * time.Minute,
```

Comment:
dont we have this as 15min in other places?

Reply:
I think we discussed and settled on only having a 15 minute wait while installing, I guess it does make sense to have the same white deletions as well. Not sure if updates should be the same or different